### PR TITLE
pnpm: stage0 bootstrap package

### PR DESCRIPTION
**Pull request by ajayk**
Bootstrap package for pnpmp , we will use this later to build pnpm 

depends on #5637 
#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

### CODE REVIEW

File: pnpm-stage0.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  contents:
+    packages:
+      - wolfi-base
+      - build-base
+      - nodejs-18
+      - ca-certificates-bundle
+      - curl
+      - git
+      - npm
+      - gcc-12
```
Comment (pdeslaur) [inline pnpm-stage0.yaml:24]: I'm surprised we need `gcc` to build pnpm!

File: pnpm-stage0.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      ln -sf /usr/lib/node_modules/pnpm/bin/pnpm.cjs  "${{targets.destdir}}"/usr/bin/pnpm
+      ln -sf /usr/lib/node_modules/pnpm/bin/pnpx.cjs  "${{targets.destdir}}"/usr/bin/pnpx
+
+update:
+  enabled: false # don't auto update stage 0 packages
+  github:
+    identifier: pnpm/pnpm
+    strip-prefix: v
+    tag-filter: v
+    use-tag: true
```
Comment (pdeslaur) [inline pnpm-stage0.yaml:45]: Should we leave that out, or is it here for documentation?

**Review by pdeslaur — APPROVED** [review]
Happy to see pnpm!

File: pnpm-stage0.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  version: 8.7.4
+  epoch: 0
+  description: "Fast, disk space efficient package manager"
+  copyright:
+    - license: MIT
+  dependencies:
+    runtime:
+      - nodejs-18
+    provides:
+      - pnpm=8.7.4
```
[the remark this answers is not in the document]
Comment (ajayk) [reply pnpm-stage0.yaml:12]: its similar to [maven-stage0](https://github.com/wolfi-dev/os/blob/main/maven-stage0.yaml) and [maven](https://github.com/wolfi-dev/os/blob/main/maven.yaml) bootstrapping

File: pnpm-stage0.yaml (hunk 1 above)
Comment (ajayk) [reply pnpm-stage0.yaml:24]: node-gyp depends on gcc , I just kept it here once we turn on the package tests , as it required for tests 

https://github.com/pnpm/pnpm/blob/main/CONTRIBUTING.md?plain=1#L31

**Pull request merged**